clover-core/src/main/java/com/atlassian/clover/util/JavaEnvUtils.java

```diff
@@ -20,6 +23,7 @@ public class JavaEnvUtils {
             javaVersion = JAVA_8;
             Class.forName("java.lang.StackWalker");
             javaVersion = JAVA_9;
+            javaVersion = SourceVersion.latestSupported().toString().substring("RELEASE_".length());
```

Comment:
I would like to have code detection based on the Java API. Therefore, please find some classes, preferably from java.core, which have been introduced in Java 10 and Java 11, respectively.

Reply:
I have changed the way to get the Java runtime version to use classes/methods introduced in each version.